File: hashmap/1. Two Sum.py

from typing import Dict, List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map: Dict[int, int] = {}
        for ind, value in enumerate(nums):
            map[value] = ind
        for ind, value in enumerate(nums):
            remain = target - value
            if map.get(remain) != None and map.get(remain) != ind:
                return [ind, map[remain]]
        return []
    

File: hashmap/1. Two Sum/1. Two Sum.go
package main

func TowSum(nums []int, target int) [2]int {

	var obj = map[int]int{}

	for ind, no := range nums {
		remain := target - no
		if i, ok := obj[remain]; ok == true {
			return [2]int{ind, i}
		}
		obj[no] = ind
	}

	return [2]int{}
}
